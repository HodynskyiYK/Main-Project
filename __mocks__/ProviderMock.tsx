import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}