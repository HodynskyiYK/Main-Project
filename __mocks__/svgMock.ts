// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock