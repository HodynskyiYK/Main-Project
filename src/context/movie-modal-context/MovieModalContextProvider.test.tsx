import React from 'react'
import { render } from '@testing-library/react'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('MovieModalContextProvider tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock><div>Children</div></ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})