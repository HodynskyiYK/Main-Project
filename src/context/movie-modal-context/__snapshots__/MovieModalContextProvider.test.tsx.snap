// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieModalContextProvider tests To match snapshot 1`] = `
<DocumentFragment>
  <div>
    Children
  </div>
</DocumentFragment>
`;
